test/src/jvmTest/kotlin/io/realm/test/compiler/IndexedTests.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.test.compiler
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import io.realm.interop.CollectionType
+import io.realm.util.CompilerTest.compileFromSource
+import io.realm.util.TypeDescriptor.allFieldTypes
+import org.junit.Test
+import kotlin.reflect.KClassifier
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class IndexedTests {
+    @Test
+    fun `indexed supportness`() {
```

Comment:
Shouldn't we also have a test for the opposite? i.e links and link-lists do not support it.

Also, we should define the behavior for fields with `@PrimaryKey`... I would be tempted to say we just ignore it there, since it doesn't really do anything.

But just to make sure we verify the behaviour if (when) it changes again ;)